path_roles.go

```diff
@@ -162,16 +195,34 @@ func (b *tfBackend) pathRolesWrite(ctx context.Context, req *logical.Request, d
 	}
 
 	roleEntry.Name = name
-	if organization, ok := d.GetOk("organization"); ok {
-		roleEntry.Organization = organization.(string)
+
+	// Will set if users dont. Must be set for multi-team tokens
+	if credentialTypeRaw, ok := d.GetOk("credential_type"); ok {
+		roleEntry.CredentialType = credentialTypeRaw.(string)
+		if !strutil.StrListContains(credentialType_Values(), roleEntry.CredentialType) {
+			return logical.ErrorResponse("unrecognized credential type: %s", roleEntry.CredentialType), nil
+		}
 	}
 
-	if teamID, ok := d.GetOk("team_id"); ok {
-		roleEntry.TeamID = teamID.(string)
+	if organization, ok := d.GetOk("organization"); ok {
+		roleEntry.Organization = organization.(string)
+		if roleEntry.CredentialType == "" {
+			roleEntry.CredentialType = organizationCredentialType
+		}
 	}
 
 	if userID, ok := d.GetOk("user_id"); ok {
 		roleEntry.UserID = userID.(string)
+		if roleEntry.CredentialType == "" {
+			roleEntry.CredentialType = userCredentialType
+		}
+	}
+
+	if teamID, ok := d.GetOk("team_id"); ok {
+		roleEntry.TeamID = teamID.(string)
+		if roleEntry.CredentialType == "" {
+			roleEntry.CredentialType = teamLegacyCredentialType
```

Comment:
Same thing here as a previous comment, should it check both `r.CredentialType == "" || r.CredentialType == teamLegacyCredentialType` to catch cases where users set it explicitly to `team_legacy`?

Reply:
good catch also!